magento/magento2
# magento/magento2#35706 Remote storage issue #35706

### Description (*)

- Created new ImageFactory class that is refactored service class Zend_Pdf_Resource_ImageFactory for using external S3 Storage images in invoice

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#35706

### Manual testing scenarios (*)

1. Install any S3 compatible storage (locally)
1.1 
```
wget https://dl.min.io/server/minio/release/linux-amd64/minio
chmod +x minio
```
1.2 Start MinIO. For example: 
`MINIO_ROOT_USER=monteshot MINIO_ROOT_PASSWORD=monteshot ./minio server /home/monteshot/minio/magento --console-address ":9901" --address ":9801"`

1.3 Create your bucket
![image](https://user-images.githubusercontent.com/2836146/186894718-17b76aa9-43a5-4033-b20e-f0e4214f743b.png)
1.4 Connect Application to S3 Storage
`bin/magento setup:config:set \
    --remote-storage-driver="aws-s3" \
    --remote-storage-endpoint="http://127.0.0.1:9801" \
    --remote-storage-region="ua-shostka-1" \
    --remote-storage-bucket="magecd" \
    --remote-storage-key="magecd12" \
    --remote-storage-secret="magecd12" \
    --no-interaction`
1.5 Run Sync
`bin/magento remote-storage:sync
`
2. Go to configuration and add image as discribed in Task 
![image](https://user-images.githubusercontent.com/2836146/186895141-611c814c-ba8a-4e46-861a-b41fd2440187.png)
https://github.com/magento/magento2/issues/35706#issue-1288848631
3.
Check the PDF invoice w/o exception
![image](https://user-images.githubusercontent.com/2836146/186895380-2452feb4-9adc-49c7-9f2d-e298a56bb310.png)

### Questions or comments


### Contribution checklist (*)
